CircuitPython_Pyloton/code.py

```diff
@@ -0,0 +1,59 @@
+from time import time
+import adafruit_ble
+import board
+import pyloton
+
+# PyLint can't find BLERadio for some reason so special case it here.
+ble = adafruit_ble.BLERadio()    # pylint: disable=no-member
+
+display = board.DISPLAY
+
+
+heart = True
+speed = True
+cad = True
+ams = True
+debug = False
+
+# 84.229 is wheel circumference (700x23 in my case)
+pyloton = pyloton.Pyloton(ble, display, 84.229, heart, speed, cad, ams, debug)
+
+pyloton.show_splash()
+
+ams = pyloton.ams_connect()
+
+
+start = time()
+hr_connection = None
+speed_cad_connections = []
+radio = None
+while True:
+    if heart:
+        if not hr_connection:
+            print("Running hr_connection")
+            hr_connection = pyloton.heart_connect()
+            ble.stop_scan()
+    if speed or cad:
+        if not speed_cad_connections:
+            print("Running speed_cad_connection")
+            speed_cad_connections = pyloton.speed_cad_connect()
+
+    if time()-start >= 45:
+        pyloton.timeout()
+        break
+    # Stop scanning whether or not we are connected.
+    ble.stop_scan()
+
+    # You may need to remove some parts of the following 2 lines depending on what
+    # devices you are using.
```

Comment:
Need to explicate this more.

Reply:
@dhalbert How does this sound?
``` 
# You may need to remove some parts of the following line depending on what devices you are using.
# For example, remove hr_connection and hr_connection.connected if you aren't using a heart rate
# monitor. Do the same for other sensors.
```

Reply:
You could do this in code by testing whether `heart` is `True` rather than by asking the user to edit the code.